src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/UseNameOfInPlaceOfStringTests.Fixer.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.CSharp.Analyzers.Maintainability;
+using Microsoft.CodeQuality.VisualBasic.Analyzers.Maintainability;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeQuality.Analyzers.Maintainability.UnitTests
+{
+    public class UseNameOfInPlaceOfStringTests : CodeFixTestBase
+    {
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
+        {
+            return new UseNameofInPlaceOfStringAnalyzer();
+        }
+
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new UseNameofInPlaceOfStringAnalyzer();
+
+        }
+
+        protected override CodeFixProvider GetBasicCodeFixProvider()
+        {
+                return new UseNameOfInPlaceOfStringFixer();
+        }
+
+        protected override CodeFixProvider GetCSharpCodeFixProvider()
+        {
+                return new UseNameOfInPlaceOfStringFixer();
+        }
+
+        [Fact]
+        public void Fixer_CSharp_ArgumentMatchesAParameterInScope()
+        {
+            VerifyCSharpFix(@"
+using System;
+class C
+{
+    void M(int x)
+    {
+        throw new ArgumentNullException(""x"");
```

Comment:
Can you add a test where there are inline comments, say `throw new ArgumentNullException(/*Leading*/"x"/*Trailing*/);`? Code fix should not remove comments.

Reply:
Yes, test was added